core/src/main/java/org/elasticsearch/common/inject/SpawnShittyModules.java

```diff
@@ -20,9 +20,19 @@
 package org.elasticsearch.common.inject;
 
 /**
+ * This interface can be added to a Module to spawn sub modules. DO NOT USE.
  *
+ * This is fundamentally broken.
+ * <ul>
+ * <li>If you have a plugin with multiple modules, return all the modules at once.</li>
+ * <li>If you are trying to make the implementation of a module "pluggable", don't do it.
+ * This is not extendable because custom implementations (using onModule) cannot be
+ * registered before spawnModules() is called.</li>
+ * </ul>
+ * @deprecated Create modules directly
  */
-public interface SpawnModules {
+@Deprecated
```

Comment:
> I'd remove this change from this PR and I'm all for considering another PR that really tackles the problem here, providing a good alternative that still promotes modularity.

uh, when you find a crappy/troublesome api or functionality, there need not always be a replacement. Sometimes things are just a bad idea and should be removed completely.